### ISSUE TITLE
Dolby Vision, Atmos, DTS:X not identified

### ISSUE THREAD

**Issue by C4Wiz** [MediaBrowser/Emby#3691]
i have a MKV with dolby vision and atmos audio that is id'd as hdr and truehd:

<img width="666" alt="Screen Shot 2021-06-23 at 7 54 10 AM" src="https://user-images.githubusercontent.com/3883521/123092241-494c0100-d3f8-11eb-9d1e-4cca83c6febb.png">

```
General
Unique ID                                : 124463260808362223037818789294941015865 (0x5DA2C08968A04E5CB11E90DB97AE0F39)
Complete name                            : /Volumes/Video/Movies/Annihilation (2018)-2160p/Annihilation (2018)_TrueHD Atmos.Dolby Vision.mkv
Format                                   : Matroska
Format version                           : Version 2
File size                                : 43.4 GiB
Duration                                 : 1 h 55 min
Overall bit rate mode                    : Variable
Overall bit rate                         : 54.1 Mb/s
Movie name                               : Annihilation
Encoded date                             : UTC 2020-09-20 13:44:50
Writing application                      : MakeMKV v1.15.2 linux(x64-release)
Writing library                          : libmakemkv v1.15.2 (1.3.10/1.5.2) x86_64-linux-gnu

Video
ID                                       : 1
ID in the original source medium         : 4113 (0x1011)
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main 10@L5.1@High
HDR format                               : Dolby Vision, Version 1.0, dvhe.07.06, BL+EL+RPU, Blu-ray compatible / SMPTE ST 2086, HDR10 compatible
Codec ID                                 : V_MPEGH/ISO/HEVC
Duration                                 : 1 h 55 min
Bit rate                                 : 50.0 Mb/s
Width                                    : 3 840 pixels
Height                                   : 2 160 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0 (Type 2)
Bit depth                                : 10 bits
Bits/(Pixel*Frame)                       : 0.251
Stream size                              : 40.2 GiB (92%)
Language                                 : English
Default                                  : No
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.2020
Transfer characteristics                 : PQ
Matrix coefficients                      : BT.2020 non-constant
Mastering display color primaries        : Display P3
Mastering display luminance              : min: 0.0001 cd/m2, max: 1000 cd/m2
Original source medium                   : Blu-ray

Audio
ID                                       : 2
ID in the original source medium         : 4352 (0x1100)
Format                                   : MLP FBA 16-ch
Format/Info                              : Meridian Lossless Packing FBA with 16-channel presentation
Commercial name                          : Dolby TrueHD with Dolby Atmos
Codec ID                                 : A_TRUEHD
Duration                                 : 1 h 55 min
Bit rate mode                            : Variable
Bit rate                                 : 4 062 kb/s
Maximum bit rate                         : 7 278 kb/s
Channel(s)                               : 8 channels
Channel layout                           : L R C LFE Ls Rs Lb Rb
Sampling rate                            : 48.0 kHz
Frame rate                               : 1 200.000 FPS (40 SPF)
Bit depth                                : 24 bits
Compression mode                         : Lossless
Stream size                              : 3.26 GiB (8%)
Title                                    : Surround 7.1
Language                                 : English
Default                                  : Yes
Forced                                   : No
Original source medium                   : Blu-ray
Number of dynamic objects                : 11
Bed channel count                        : 1 channel
Bed channel configuration                : LFE

Menu
00:00:00.000                             : en:Chapter 01
00:05:08.975                             : en:Chapter 02
00:11:39.156                             : en:Chapter 03
00:17:06.483                             : en:Chapter 04
00:24:08.405                             : en:Chapter 05
00:28:59.446                             : en:Chapter 06
00:37:24.200                             : en:Chapter 07
00:41:59.475                             : en:Chapter 08
00:49:11.281                             : en:Chapter 09
00:56:06.404                             : en:Chapter 10
01:03:22.131                             : en:Chapter 11
01:10:24.720                             : en:Chapter 12
01:17:21.928                             : en:Chapter 13
01:22:20.226                             : en:Chapter 14
01:30:25.962                             : en:Chapter 15
01:38:48.589                             : en:Chapter 16
01:43:58.357                             : en:Chapter 17
01:48:14.279                             : en:Chapter 18
```

**Comment by softworkz** [comment]
ffmpeg does not have support for Dolby Vision nor Dolby Atmos, that's why it only indicates the compatible subsets: TrueHD and HDR10.

**Comment by C4Wiz** [comment]
But it plays in dolby vision with atmos

David Smythe
Sent from my  iPhone 12 Pro Max on iOS 14

On Jun 23, 2021, at 12:28 PM, softworkz ***@***.***> wrote:

﻿

ffmpeg does not have support for Dolby Vision nor Dolby Atmos, that's why it only indicates the compatible subsets: TrueHD and HDR10.

**Comment by softworkz** [comment]
> But it plays in dolby vision with atmos

Yes, in case of DirectPlay, when the client is able to handle these format.

But for streams detection, we can only detect what ffmpeg/ffprobe supports,

**Comment by C4Wiz** [comment]
Can you add support for file naming to populate the icons?

David Smythe
Sent from my  iPhone 12 Pro Max on iOS 14

On Jun 23, 2021, at 12:45 PM, softworkz ***@***.***> wrote:

﻿

But it plays in dolby vision with atmos

Yes, in case of DirectPlay, when the client is able to handle these format.

But for streams detection, we can only detect what ffmpeg/ffprobe supports,

**Comment by softworkz** [comment]
> Can you add support for file naming to populate the icons?

Which icons?

**Comment by C4Wiz** [comment]
> > Can you add support for file naming to populate the icons?
> 
> Which icons?

For dolby vision, atmos, dts:x, hdr10+ From r the client app on shield

**Comment by softworkz** [comment]
The detection is done by the server and it works by analyzing file content, not file names. The detection results are not only used for client display; they play an important role in the decision-process for playback (what a client is capable to process and whether the server would need to do some conversion).  File naming is unreliable, often incomplete and there's no standardization anyway.
I'm sure that these things will be added to ffmpeg sooner or later, but right now, I can't think of any reasonable way to detect these things.

**Comment by C4Wiz** [comment]
Make a list of file names that trigger flags and post them, not a hard thing to do.

**Comment by softworkz** [comment]
> The detection results are not only used for client display; they play an important role in the decision-process for playback

**Comment by C4Wiz** [comment]
I get that but why can file naming not be separate for flags?

**Comment by softworkz** [comment]
Sure - it could be, yes. 

But we don't have such kind of "flags" mechanism.

> not a hard thing to do

Add database fields for it, add that functionality to almost 20 client applications, decide what to display when file name info differs from content analysis, respond to user questions about why "flags" are not matching content, how about multiple audio streams, how would the file names have to be like to convey that information, which users would be willing to keep renaming their files to make use of that "feature"?

I understand that this might appear to be a useful idea at first sight...
..but it doesn't make the slightest sense, once you think it through to the end in all ways.

**Comment by C4Wiz** [comment]
> Sure - it could be, yes.
> 
> But we don't have such kind of "flags" mechanism.
> 
> > not a hard thing to do
> 
> Add database fields for it, add that functionality to almost 20 client applications, decide what to display when file name info differs from content analysis, respond to user questions about why "flags" are not matching content, how about multiple audio streams, how would the file names have to be like to convey that information, which users would be willing to keep renaming their files to make use of that "feature"?
> 
> I understand that this might appear to be a useful idea at first sight...
> ..but it doesn't make the slightest sense, once you think it through to the end in all ways.

i understand yes. would it not be more suitable to use mediainfo instead of ffmpeg to detect stream info?

**Comment by C4Wiz** [comment]
perhaps a easier solution would be to add the ability to edit those fields under edit metadata, strictly for appearences.

**Comment by softworkz** [comment]
> i understand yes. would it not be more suitable to use mediainfo instead of ffmpeg to detect stream info?

No - the major difference is that MediaInfo results are primarily intended for display. ffprobe detection results are using the exact same string values that ffmpeg understands and that we can use as parameters for transcoding command lines.
These components are building up some kind of "microcosm" within Emby Server is operating and adaptively serving your media. All things need to fit together precisely, otherwise it would break.

About establishing an additional set of "Display-Metadata": I don't think that this would provide a real benefit, but it would surely create a lot of confusion.

I'm really afraid, but the only realistic solution that I can see is to wait until ffmpeg/ffprobe gets support for these things.

**Comment by C4Wiz** [comment]
so you support dolby vision, atmos and dtsx for playback on the shieldtv pro, but you don't support anyway to know the current movie contains any of the above?

am i the only one that sees a issue with that?

**Comment by softworkz** [comment]
> so you support dolby vision, atmos and dtsx for playback on the shieldtv pro,

No. It's the Shield itself or the TV that supports these things. 

> but you don't support anyway to know the current movie contains any of the above?

Yes, that's the current situation. Emby doesn't understand Atmos and Vision, only the compatible subsets (TrueHD, HDR10).
In case of direct play, it sends the original streams to the client, and it depends on the client what it is able to do. Emby does not know about that, it only knows for example, that there is an HDR10 video and it can direct-stream the video because the client indicates support for HDR10. 
When the video stream is not only HDR10 but also DVision AND the client is capable to process it, then the client might present the video with DolbyVision, but that's already out-of-reach from Emby Server.

**Comment by C4Wiz** [comment]
so you are not will to do. anything on the matter?

i don't see why adding a few editable fields in the edit metadata window is such a big deal, it's purely astetic. and from a collection standpoint it just makes for a more complete package. it's nice to know what you are actually gonna be seeing and hearing.

**Comment by softworkz** [comment]
> it's nice to know what you are actually gonna be seeing and hearing.

Well, that's the point: you won't be able to know what you are going to see and hear because it depends on the client's capabilities and when the client doesn't support DV or Atmos, then you won't get that even though it's shown in the media info.
That means, it would be bad either way.

> so you are not will to do. anything on the matter?

Let's see what @LukePulverenti thinks.

**Comment by C4Wiz** [comment]
Embyserver for me is nothing more then a front end for shield pro 2019’s on tvs and avrs that will play everything, so 0 transcoding. I use TinyMediaManager for metadata creation

**Comment by C4Wiz** [comment]
anyone?

**Comment by LukePulverenti** [comment]
Hi, HDR and Dolby Vision detection has been added to the upcoming 4.8 server release. We'll look at Atmos and DTS:X shortly after that. Thanks.
